tutorials/nservicebus-101/lesson-2/tutorial.md

```diff
@@ -0,0 +1,190 @@
+---
+title: "NServiceBus 101 Lesson 2: Sending a command"
+---
+
+Sending and receiving messages is at the core of any NServiceBus system. Durable messages passed between processes allow reliable communication between those processes, even if one of them is temporarily unavailable. In this lesson we'll show how to send and process a message.
+
+
+## Objectives
+
+By the end of this lesson, you will have learned how to:
+
+ * Define messages
+ * Define message handlers
+ * Send and receive a message locally
+ * Use NServiceBus's logging capabilities
+
+
+## What is a message?
+
+A [**message**](/nservicebus/messaging/messages-events-commands.md) is a collection of data sent via one-way communication between two endpoints. In NServiceBus, we define message via simple classes.
+
+In this lesson, we'll focus on [commands](/nservicebus/messaging/messages-events-commands.md#command). In [Lesson 4: Publishing events](../lesson-4/) we'll expand to look at events as well.
+
+Defining a command is pretty easy. We just create a class and mark it with the `ICommand` marker interface.
+
+snippet:Command
+
+The marker interface has no implementation, but lets NServiceBus know that the class is a command so that it can build up some metadata about the message type when the endpoint starts up. Any properties you create within the message constitute the message data.
+
+The name of the command class is also important. A command is an order to do something, so it should be named in the [imperative tense](https://en.wikipedia.org/wiki/Imperative_mood). `PlaceOrder`, `UpdateCustomerStatus`, and `ChargeCreditCard` are all great names for commands, because they are phrased as a command and are very specific. You can easily guess exactly what each of those messages will do. `UserMessage`, on the other hand, is not a good example. It is not in the imperative, and it's not very specific either. Another developer should know exactly what a command's purpose is just by reading the name.
+
+Command names should also convey business intent. `UpdateCustomerPropertyXYZ` isn't a good name for an command either, because it's focused only on the data manipulation, rather than the business meaning behind it. `MarkCustomerAsGold`, or something else that is more business-oriented, is probably a better choice.
+
+When sending a message, the endpoint's [message serializer](/nservicebus/serialization/) will serialize an instance of the `DoSomething` class and add that to the contents of the outgoing message that goes to the queue. On the other end, the receiving endpoint will deserialize the message from its serialized form back to an instance of the message class so that it can be used in code.
+
+Messages can support whatever data the serializer can successfully process. Thus messages can even contain child objects or collections.
```

Comment:
perhaps a comment on the supported range of structures is dictated by the choice of serializer https://docs.particular.net/nservicebus/serialization/#supported-serializers